### PR TITLE
storage of water vapor quality mask

### PR DESCRIPTION
**Pull request by HolgerPollyNet**
please review! Then i can change all the other main functions of diffeent polly systems

### CODE REVIEW

File: config/template_pollynet_processing_chain_link.txt [hunk 1]
```diff
@@ -2,3 +2,5 @@ name,start time,end time,config file,processing function, update Info(as short a
 pollyxt_dwd,2017-06-09 00:00:00,2019-06-06 00:00:00,pollyxt_dwd_config.json,pollynet_processing_chain_pollyxt_dwd.m,This is really state_of_the_art,pollyxt_dwd_defaults.json
 pollyxt_tau,2019-01-01 00:00:00,2020-01-01 00:00:00,pollyxt_tau_config_20190910.json,pollynet_processing_chain_pollyxt_tau.m,Tel Aviv,pollyxt_tau_defauts.json
 pollyxt_tropos,2019-07-02 00:00:00,2020-12-31 00:00:00,pollyxt_tropos_config_20190702.json,pollynet_processing_chain_pollyxt_tropos.m,Move to Leipzig for mantainence,pollyxt_tropos_defaults.json
+arielle,2018-10-25 00:00:00,2018-12-09 00:00:00,arielle_config_20181101.json,pollynet_processing_chain_arielle.m,PS116,arielle_defaults.json
```
Comment (ZPYin) [inline config/template_pollynet_processing_chain_link.txt:5]: Please don't add any new lines in the template file, unless you need to demonstrate something new. In your case, you should edit your personal *link.txt file. 

PS: Files with prefix of 'template_' are used for demonstration.

File: lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m [hunk 2]
```diff
@@ -55,7 +57,10 @@ function pollyxt_save_WVMR_RH(data, taskInfo, config)
 netcdf.putVar(ncID, varID_time, datenum_2_unix_timestamp(data.mTime));
 netcdf.putVar(ncID, varID_WVMR, fillmissing(data.WVMR, missing_value));
 netcdf.putVar(ncID, varID_RH, fillmissing(data.RH, missing_value));
-
+try
+    netcdf.putVar(ncID, varID_QM_WVMR, fillmissing(data.quality_mask_WVMR, missing_value));
+    netcdf.putVar(ncID, varID_QM_RH, fillmissing(data.quality_mask_RH, missing_value));
+end
```
Comment (ZPYin) [inline lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m:63]: would it be better to use "if ... else..." here? See the code below.

```matlab
if isfield('data', 'quality_mask_WVMR')
    netcdf.putVar(ncID, varID_QM_WVMR, fillmissing(data.quality_mask_WVMR, missing_value));
end

if isfield('data', 'quality_mask_RH')
    netcdf.putVar(ncID, varID_QM_RH, fillmissing(data.quality_mask_RH, missing_value));
end
```

I think it delivers clear meaning and is more friendly for future improvement.

**Comment by HolgerPollyNet** [comment]
Sure looks good, go for IT
________sure ________________________________
Von: ZPYin [notifications@github.com]
Gesendet: Donnerstag, 11. Juni 2020 15:08
An: PollyNET/Pollynet_Processing_Chain
Cc: Holger Baars; Author
Betreff: Re: [PollyNET/Pollynet_Processing_Chain] I added the storage of water vapor quality mask (#63)

@ZPYin commented on this pull request.

________________________________

In lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m<https://github.com/PollyNET/Pollynet_Processing_Chain/pull/63#discussion_r438768005>:

> @@ -55,7 +57,10 @@ function pollyxt_save_WVMR_RH(data, taskInfo, config)
 netcdf.putVar(ncID, varID_time, datenum_2_unix_timestamp(data.mTime));
 netcdf.putVar(ncID, varID_WVMR, fillmissing(data.WVMR, missing_value));
 netcdf.putVar(ncID, varID_RH, fillmissing(data.RH, missing_value));
-
+try
+    netcdf.putVar(ncID, varID_QM_WVMR, fillmissing(data.quality_mask_WVMR, missing_value));
+    netcdf.putVar(ncID, varID_QM_RH, fillmissing(data.quality_mask_RH, missing_value));
+end


would it be better to use "if ... else..." here? See the code below.

if isfield('data', 'quality_mask_WVMR')
    netcdf.putVar(ncID, varID_QM_WVMR, fillmissing(data.quality_mask_WVMR, missing_value));
end

if isfield('data', 'quality_mask_RH')
    netcdf.putVar(ncID, varID_QM_RH, fillmissing(data.quality_mask_RH, missing_value));
end

I think it delivers clear meaning and is more friendly for future improvement.

**Comment by HolgerPollyNet** [comment]
I pushed a new version, but I am not sure if the way I am doing int now is interferring wiht the 

"% re enter define mode
netcdf.reDef(ncID);"

command.

Can you review?

File: lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m [hunk 3]
```diff
@@ -53,8 +55,31 @@ function pollyxt_save_WVMR_RH(data, taskInfo, config)
...
+if isfield('data', 'quality_mask_WVMR')
+    netcdf.putVar(ncID, varID_WVMR, fillmissing(data.WVMR, missing_value));	
+    netcdf.putVar(ncID, varID_QM_WVMR, fillmissing(data.quality_mask_WVMR, missing_value));
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'unit', 'flag');
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'unit_html', 'flag');
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'long_name', 'Quality mask for WVMR ratio retrieval. 0=ok,1=SNR too low,2=depol calibation');
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'standard_name', 'QM_WVMR');
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'plot_range', config.xLim_Profi_WV_RH);
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'plot_scale', 'linear');
+    netcdf.putAtt(ncID, varID_QM_WVMR, 'source', campaignInfo.name);
```
Comment (ZPYin) [inline lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m:69]: I think the problem should be caused by the `putAtt` operations before entering define mode.

As the [MATLAB document](https://www.mathworks.com/help/matlab/ref/netcdf.redef.html#:~:text=netcdf.,a%20previous%20call%20to%20netcdf.) says: attributes can be added under define mode. Try to move the `putAtt` after `netcdf.reDef(ncID)`. Let's see what will happen.

**Comment by HolgerPollyNet** [comment]
new version online, i also fixed some bugs (latitude), please check

File: lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m [hunk 4]
```diff
@@ -51,34 +54,20 @@ function pollyxt_save_WVMR_RH(data, taskInfo, config)
 
 % write data to .nc file
 netcdf.putVar(ncID, varID_altitude, data.alt0);
-netcdf.putVar(ncID, varID_altitude, data.alt0);
+netcdf.putVar(ncID, varID_latitude, data.lat);
```
Comment (ZPYin) [inline lib/polly_general_func_lib/pollyxt_save_WVMR_RH.m:57]: Ooop, nice done...

**Comment by ZPYin** [comment]
I fixed a minor bug when testing.
Now you can pull the `dev` branch and extend the feature to all pollyxts.

**Pull request merged**